features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/data/ComposeContextLexer.kt

```diff
@@ -0,0 +1,267 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.compose.internal.data
+
+internal object ComposeContextLexer {
+
+    private sealed class TokenReader {
+
+        abstract fun process(c: Char)
+        abstract fun updateComposeContext(composeContext: ComposeContext?): ComposeContext?
+        protected var nextReader: TokenReader? = null
+
+        fun nextReader() = nextReader
+
+        protected fun nextReader(c: Char): TokenReader {
```

Comment:
side note: maybe linking some documentation in the code comments will be useful to understand the format we are dealing with